rpg/views.py

```diff
@@ -140,6 +143,10 @@ def get_context_data(self, **kwargs: dict[str, Any]):
             self.request.user.pk, self.amount, settings.CERT_HASH_KEY
         )
         context["pk"] = self.request.user.pk
+        try:
+            context["student_pk"] = infer_student(self.request).pk
+        except:
+            context["student_pk"] = -1
```

Comment:
i think you should use `None` instead of `-1`, that's more clear what is meant imo

Reply:
also you should use explicit `except Http404`, in general using a bare `except` is considered bad

Reply:
I don't think Http404 is a BaseException, so it can't be caught? I'm not sure if that's actually how it works